static/app/utils/profiling/differentialFlamegraph.tsx

```diff
@@ -1,32 +1,34 @@
 import {ColorChannels, FlamegraphTheme} from './flamegraph/flamegraphTheme';
 import {Profile} from './profile/profile';
-import {relativeChange} from './units/units';
 import {Flamegraph} from './flamegraph';
 import {FlamegraphFrame} from './flamegraphFrame';
 
-function countFrameOccurrences(
-  frames: ReadonlyArray<FlamegraphFrame>
-): Map<string, number> {
+const frameDiffKey = (frame: FlamegraphFrame) => {
+  return (
+    frame.frame.name +
+    (frame.frame.file ?? '') +
+    (frame.frame.path ?? '') +
+    (frame.frame.module ?? '') +
+    (frame.frame.package ?? '')
+  );
+};
+
+function makeFrameMap(frames: ReadonlyArray<FlamegraphFrame>): Map<string, number> {
   const counts = new Map<string, number>();
 
   for (const frame of frames) {
-    const key = frame.frame.name + (frame.frame.file ? frame.frame.file : '');
-
-    if (counts.has(key)) {
-      counts.set(key, counts.get(key)! + 1);
-    } else {
-      counts.set(key, 1);
-    }
+    counts.set(frameDiffKey(frame), frame.node.selfWeight);
```

Comment:
DId you mean to sum all the weights? This will over write the weight each time the same frame is seen.